spec/light-client/supervisor/supervisor_001_draft.md

```diff
@@ -463,69 +463,94 @@ func Sequential-Supervisor (initdata LCInitData) (Error) {
-> block. However, if a peer provides a conflicting
-> lightblock, the question is to distinguish the case of a
-> [bogus](https://informal.systems) block (upon which operation should proceed) from a
-> [light client attack](https://informal.systems) (upon which operation should stop). In
-> case of a bogus block, the lightclient might be forced to do
-> backwards verification until the blocks are out of the trusting
-> period, to make sure no previous validator set could have generated
-> the bogus block, which effectively opens up a DoS attack on the lightclient
-> without adding effective robustness.
+trust the initial data given to it by the user. It cannot perform any
+detection of an attack yet instead requires an initial point of trust.
+There are three forms of initial data:
+
+- A trusted block from a prior initialization
+- A trusted height and hash
+- A genesis file
+
+The golang light client implementation checks this initial data in that 
+order; first attempting to find a trusted block from the trusted store, 
+then acquiring a light block from the primart at the trusted height and matching
+the hash, or finally checking for a genesis file to verify the initial header.
+
+The light client doesn't need to check if the trusted block is within the
+trusted period because it already trusts it, however, if the light block is 
+outside the trust period, there is a higher chance the light client won't be
+able to verify anything. 
+
+Cross-checking this trusted block with providers upon initialization is helpful
+for ensuring liveness and correct configuration of such nodes but does not 
+increase trust since proving a conflicting block is a 
+[light client attack](https://informal.systems) 
+and not just a [bogus](https://informal.systems) block could result in
+performing backwards verification beyond the trusted period, thus a fruitless
+endeavour.
+
+However, with the notion of it's better to fail earlier than later, the golang
+light client implementation will perform a consistency check on all providers
+and will error if one returns a different header, allowing the user
+the opportunity to reinitialize.
 
 #### **[LC-FUNC-INIT.1]:**
 
 ```go
-func InitLightClient (initData LCInitData) (LightStore, Error) {
+func InitLightClient(initData LCInitData) (LightStore, Error) {
+    var initialBlock LightBlock
 
-    if LCInitData.LightBlock != nil {
-        // we trust the provided initial block.
-        newblock := LCInitData.LightBlock
-    }
-    else {
-        genesisBlock := makeblock(initData.genesisDoc);
+    switch {
+    case LCInitData.TrustedBlock != nil:
+        // we trust the block from a prior initialization
+        initialBlock = LCInitData.TrustedBlock
 
-        result := NoResult;
-        while result != ResultSuccess {
-            current = FetchLightBlock(PeerList.primary(), genesisBlock.Header.Height + 1)
-            // QUESTION: is the height with "+1" OK?
+    case LCInitData.TrustedHash != nil:
```

Comment:
Please add the Field `TrustedHash` to the data structure definition further above in the code.

Reply:
Sure. I've added them. One might argue that the `TrustedHash` and `TrustedHeight` combination are just for convenience (instead of passing the entire `TrustedBlock`) and so doesn't belong in the spec. But perhaps it's good to have them here just for documentation sake because it's definitely a prevalent concept in the golang codebase

Reply:
I think it is good to give the options and details here. Thanks for adding the hash option?